File: packages/ruby/lib/opensms/resources/lookups.rb
# frozen_string_literal: true

require_relative "base"

module Opensms
  module Resources
    # The +lookups+ resource: number lookups (country, carrier, porting).
    # Accessed as +client.lookups+.
    class Lookups < Base
      # POST /v1/lookup -> 200 (completed) or 202 (pending) Lookup.
      #
      # @param params [Hash] :to (E.164, required)
      # @return [Hash] Lookup
      def create(params = nil, idempotency_key: nil, **kwargs)
        body = Models.build(merge(params, kwargs), %i[to], required: %i[to])
        @transport.request(:post, "/v1/lookup", body: body, idempotency_key: idem(idempotency_key))
      end

      # GET /v1/lookup/{id} -> Lookup.
      def get(id)
        http_get(path("/v1/lookup", id!(id)))
      end
    end
  end
end

File: packages/python/src/opensms/resources/lookups.py
"""The ``lookups`` resource."""

from __future__ import annotations

from typing import Optional

from .._serialize import path_id
from ..models import Lookup
from ._base import Resource


class Lookups(Resource):
    """Number lookups (HLR). Accessed as ``client.lookups``."""

    def create(self, *, to: str, idempotency_key: Optional[str] = None) -> Lookup:
        """Look up a number. Returns the completed lookup, or a pending one
        (``state`` ``queued``) to poll with :meth:`get`."""
        return self._t.request(
            "POST", "/v1/lookup", json_body={"to": to}, idempotent=True, idempotency_key=idempotency_key
        )

    def get(self, id: str) -> Lookup:
        """Fetch a lookup."""
        return self._t.request("GET", f"/v1/lookup/{path_id(id)}")
